.devcontainer/reinstall-cmake.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
```

Comment:
This file is not necessary for this example and adds a lot of extra complexity. The C++ code does not use CMake and works just fine with the both the base C++ extension and the C++ extension pack (which includes the CMake extension... but not CMake...).

Reply:
Vcpkg's requirements in some cases require a more recent version of cmake than is in the distro itself. This was added to the template when Vcpkg was added to the image. I think Debian 11 has a recent enough cmake though, so we could remove it.

https://github.com/devcontainers/images/blob/main/src/cpp/.devcontainer/Dockerfile#L11-L22

Reply:
The example source code does not use vcpkg either though.

Reply:
The image is general purpose not specific to the sample. The README at the top of devcontainer.json points to the general purpose template.  Happy to take a PR to illustrate its use if you would like.